XenAdmin/Wizards/PatchingWizard/PatchingWizard_UploadPage.cs

```diff
@@ -509,6 +509,12 @@ private void singleAction_Completed(ActionBase sender)
                                 AllIntroducedPoolUpdates.Add(PoolUpdate, SelectedNewPatchPath);
                             }
                         }
+
+                        foreach( var dict in (action as UploadSupplementalPackAction).SrUploadedUpdates)

```

Comment:
As we are casting action to UploadSupplementalPackAction a few times now in this block, we could consider replacing the type check `if (action is UploadSupplementalPackAction)` on line 489 with cast and check for null, like 
```
var uploadAction = action as UploadSupplementalPackAction;
if (uploadAction != null)
```